File: Month01_Rust/src/incoming/execution_model.rs
#[repr(u8)]
#[derive(PartialEq, Copy, Clone, Debug)]
pub enum ExecutionModel {
    OneOff = 0,      // Single fire-and-forget unit
    Scheduled = 1,   // Time-triggered (cron/delayed)
    EventDriven = 2, // Triggered by domain/infrastructure event
    Batch = 3,       // Large dataset processing
    Stream = 4,      // Long-lived continuous processing
    Workflow = 5,    // DAG / multistep orchestration
    Actor = 6,       // Key-partitioned stateful unit
}

impl ExecutionModel {
    const EXECUTION_MODELS: [ExecutionModel; 7] = [
        ExecutionModel::OneOff,
        ExecutionModel::Scheduled,
        ExecutionModel::EventDriven,
        ExecutionModel::Batch,
        ExecutionModel::Stream,
        ExecutionModel::Workflow,
        ExecutionModel::Actor,
    ];

    pub fn get_all() -> &'static [ExecutionModel] {
        &Self::EXECUTION_MODELS
    }
}

File: 04_Haskell/workers/src/ExecutionModel.hs
module ExecutionModel
  ( ExecutionModel (..),
  )
where

data ExecutionModel
  = OneOff -- Single fire-and-forget unit
  | Scheduled -- Time-triggered (cron/delayed)
  | EventDriven -- Triggered by domain/infrastructure event
  | Batch -- Large dataset processing
  | Stream -- Long-lived continuous processing
  | Workflow -- DAG / multistep orchestration
  | Actor -- Key-partitioned stateful unit
  deriving (Show, Eq, Ord, Enum, Bounded)